main.c

```diff
@@ -149,6 +151,22 @@ static void acquire_sleep_lock(void) {
 	sd_bus_message_unref(msg);
 }
 
+static void set_idle_hint(bool hint) {
+	swayidle_log(LOG_DEBUG, "SetIdleHint %d", hint);
+	sd_bus_message *msg = NULL;
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+	int ret = sd_bus_call_method(bus, "org.freedesktop.login1",
+			session_name, "org.freedesktop.login1.Session", "SetIdleHint",
+			&error, &msg, "b", hint);
```

Comment:
FYI, passing `NULL` instead of `&msg` should have worked, too.

Reply:
Hm, it liiks like that would leak the reply. But that should be a bug in sd-bus itself.